fmskill/compare.py

```diff
@@ -389,6 +402,119 @@ def _parse_by(self, by, n_models, n_obs):
         return by
 
+    def spatial_skill(
+        self,
+        bins=5,
+        binsize: float = None,
+        retbins: bool = False,
+        by: Union[str, List[str]] = None,
+        metrics: list = None,
+        n_min: int = None,
+        model: Union[str, int, List[str], List[int]] = None,
+        observation: Union[str, int, List[str], List[int]] = None,
+        start: Union[str, datetime] = None,
+        end: Union[str, datetime] = None,
+        area: List[float] = None,
+        df: pd.DataFrame = None,
+    ):
+        """Aggregated spatial skill assessment of model(s) on a regular spatial grid.
+
+        Parameters
+        ----------
+        bins: int, list of scalars, or IntervalIndex, or tuple of, optional
+            criteria to bin x and y by, argument bins to pd.cut(), default 5
+            define different bins for x and y a tuple
+            e.g.: bins = 5, bins = (5,[2,3,5])
+        binsize : float, optional
+            bin size for x and y dimension, overwrites bins
+            creates bins with reference to round(mean(x)), round(mean(y))
+        by : (str, List[str]), optional
+            group by column name or by temporal bin via the freq-argument
+            (using pandas pd.Grouper(freq)),
+            e.g.: 'freq:M' = monthly; 'freq:D' daily
+            by default ["model","observation"]
+        metrics : list, optional
+            list of fmskill.metrics, by default [bias, rmse, urmse, mae, cc, si, r2]
+        n_min : int, optional
+            minimum number of observations
+        model : (str, int, List[str], List[int]), optional
+            name or ids of models to be compared, by default all
+        observation : (str, int, List[str], List[int])), optional
+            name or ids of observations to be compared, by default all
+        start : (str, datetime), optional
+            start time of comparison, by default None
+        end : (str, datetime), optional
+            end time of comparison, by default None
+        area : list(float), optional
+            bbox coordinates [x0, y0, x1, y1],
+            or polygon coordinates [x0, y0, x1, y1, ..., xn, yn],
+            by default None
+        df : pd.dataframe, optional
+            user-provided data instead of the comparers own data, by default None
+
+        Returns
+        -------
+        xr.Dataset
+            skill assessment as a dataset
+
+        See also
+        --------
+        skill
+            a method for aggregated skill assessment
+
+        """
+
+        metrics = self._parse_metric(metrics)
+
+        df = self.sel_df(
+            model=model, observation=observation, start=start, end=end, area=area, df=df
+        )
+
+        df = self._add_spatial_grid_to_df(df=df, bins=bins, binsize=binsize)
+
+        n_models = len(df.model.unique())
+        n_obs = len(df.observation.unique())
+        by = self._parse_by(by, n_models, n_obs)
+        if not "xBin" in by:
```

Comment:
Would it not be better to have xBin and yBin first in the by-list (instead of appending)? It makes most sense to me, but maybe I am missing something?

Reply:
Could `xBin` and `yBin` not end up in the middle of the resulting `by` list in some scenarios?

Reply:
Could you please explain?

Reply:
if not "xBin" in by:
       by.insert(0,"xBin")
    if not "yBin" in by:
       by.insert(0,"yBin")

Reply:
Sure, this suggestion works!